tools/integration-tests/tester/tests/consensus/consensus_conflict_spam_test.go

```diff
@@ -0,0 +1,185 @@
+			}
+			prevGoF = metadata.GradeOfFinality
+		}
+	}
+}
+
+/**
+sendPairWiseConflicts receives a list of outputs controlled by a peer with certain peer index.
+It send them all to addresses controlled by the next peer, but it does so several time to create pairwise conflicts.
+The conflicts are TX_B<->TX_A<->TX_C
+*/
+func sendPairWiseConflicts(t *testing.T, peers []*framework.Node, outputs ledgerstate.Outputs,
+	keyPairs map[string]*ed25519.KeyPair,
+	txs *[]*ledgerstate.Transaction, iteration int) {
+
+	t.Logf("send pairwise conflicts on iteration %d", iteration)
+	peerIndex := (iteration + 1) % len(peers)
+
+	// find target addresses
+	targetAddresses := determineTargets(peers, iteration)
+
+	tx1 := tests.CreateTransactionFromOutputs(t, peers[0].ID(), targetAddresses, keyPairs, outputs...)
+	tx2 := tests.CreateTransactionFromOutputs(t, peers[1].ID(), targetAddresses, keyPairs, outputs[0])
+	tx3 := tests.CreateTransactionFromOutputs(t, peers[2].ID(), targetAddresses, keyPairs, outputs[2])
+
+	*txs = append(*txs, tx1, tx2, tx3)
+
+	PostTransactions(t, peers, peerIndex, "pairwise conflicts", tx1, tx2, tx3)
+}
+
+/**
+Creates conflicts as so
+TX_A<->TX_B TX_B<->TX_C TX_C<->TX_A
+*/
+func sendTripleConflicts(t *testing.T, peers []*framework.Node, outputs ledgerstate.Outputs,
+	keyPairs map[string]*ed25519.KeyPair,
+	txs *[]*ledgerstate.Transaction, iteration int) {
+	t.Logf("send triple conflicts on iteration %d", iteration)
+
+	peerIndex := (iteration + 1) % len(peers)
+
+	// find target addresses
+	targetAddresses := determineTargets(peers, iteration)
+
+	tx1 := tests.CreateTransactionFromOutputs(t, peers[0].ID(), targetAddresses, keyPairs, outputs...)
+	tx2 := tests.CreateTransactionFromOutputs(t, peers[1].ID(), targetAddresses, keyPairs, outputs[0], outputs[1])
+	tx3 := tests.CreateTransactionFromOutputs(t, peers[2].ID(), targetAddresses, keyPairs, outputs[1], outputs[2])
+
+	*txs = append(*txs, tx1, tx2, tx3)
+
+	PostTransactions(t, peers, peerIndex, "triplet conflicts", tx1, tx2, tx3)
+}
+
+func PostTransactions(t *testing.T, peers []*framework.Node, peerIndex int, attackName string, txs ...*ledgerstate.Transaction) {
+	for i, tx := range txs {
+		newPeerIndex := (peerIndex + i) % len(peers)
+		resp, err := peers[newPeerIndex].PostTransaction(tx.Bytes())
+		t.Logf("%s: post tx %s on peer %s", attackName, tx.ID().Base58(), peers[peerIndex].Name())
+		require.NoError(t, err, "%s: There was an error posting transaction %s to peer %s",
+			attackName, tx.ID().Base58(), peers[peerIndex].Name())
+		require.Empty(t, resp.Error, "%s: There was an error in the response while posting transaction %s to peer %s",
+			attackName, tx.ID().Base58(), peers[peerIndex].Name())
+		time.Sleep(time.Second)
+	}
+}
+
+func determineTargets(peers []*framework.Node, iteration int) []*ledgerstate.Address {
+	targetIndex := (iteration + 1) % len(peers)
+	targetPeer := peers[targetIndex]
+	targetAddresses := []*ledgerstate.Address{}
```

Comment:
Turn into a `make` and don't use pointers.

Reply:
since the `i` in the loop won't start with 0 I think this way is more readable.
I'll remove the pointer after I get an answer about the bloat vs. safety question I asked you on another question.

Reply:
`make` is just more in line with go style in my opinion.
And regarding the pointers: you don't really modify the Address struct itself later on no? So removing the pointer should work.